packages/snaps-execution-environments/src/common/BaseSnapExecutor.ts

```diff
@@ -251,34 +253,47 @@ export class BaseSnapExecutor {
 
     try {
       const result = await (this.methods as any)[method](...paramsAsArray);
-      this.respond(id, { result });
+      await this.#respond(id, { result });
     } catch (rpcError) {
-      this.respond(id, {
+      await this.#respond(id, {
         error: serializeError(rpcError, {
           fallbackError,
         }),
       });
     }
   }
 
-  protected notify(requestObject: Omit<JsonRpcNotification, 'jsonrpc'>) {
+  // Awaitable function that writes back to the command stream
+  async #write(chunk: Json) {
+    return new Promise<void>((resolve, reject) => {
+      this.commandStream.write(chunk, (error) => {
+        if (error) {
+          reject(error);
+          return;
+        }
+        resolve();
```

Comment:
Should we add some comment somewhere in code about why is this done in this way? It might be confusing and look "redundant" to somebody in the future.

Reply:
The one now added is good.